### DISCUSSION TITLE
Athom KM01 serious lag with hardware switches

### DISCUSSION THREAD

**Discussion by uchristo** [arendst/Tasmota#19316] [Q&A]
Hello everybody,

I'm currently trying to exchange all my shelly 2.5pm's with athom [KM01](https://de.athom.tech/blank-1/tasmota-curtain-module) because of general instability ( [Related issue](https://github.com/arendst/Tasmota/issues/10237) )
First Problem: I need to change all the cabling because S1 and S2 on KM01 will need neutral line instead of phase like the shelly did.

After cabling this and changing the Switchmode for both switches to "2" instead of "1" it seems to work, but there's a serious noticeable lag when moving the curtain using the hardware switches (compared to shelly). I noticed the template having a Switchdebounce 300 already included in the template....
https://de.athom.tech/blank-1/tasmota-curtain-module
" {"NAME":"Athom Curtain Module","GPIO":[0,0,0,160,161,32,0,0,224,576,225,0,0,0],"FLAG":0,"BASE":18,"CMND":"SO80 1;ShutterMode 1;**SwitchDebounce 300;ButtonDebounce 300"**} "

I then decreased this to 40ms, but the issue persists. Is there a way to disable this lag? I included the whole config-Commands converted using decode-config.py (WiFi scrambled)

[hwr_commands.txt](https://github.com/arendst/Tasmota/files/12343777/hwr_commands.txt)

**Comment by sfromis** [comment]
To explore what happens, do enable `Weblog 3` and watch the console for details. If the Button pin is involved, `SetOption13` may be relevant for immediate action. If you have not allowed it to have NTP server access, that may cause slowdowns.

**Reply by uchristo** [reply]
I recently configured option13 to true (but am using switches, not buttons) ntp seems to be configured, my Firewall won't block this:
07:02:57.796 CMD: ntpserver
07:02:57.798 SRC: WebConsole from 192.168.151.53
07:02:57.801 CMD: Grp 0, Cmd 'NTPSERVER', Idx 1, Len 0, Pld -99, Data ''
07:02:57.805 RSL: RESULT = {"NtpServer1":"2.pool.ntp.org","NtpServer2":"2.europe.pool.ntp.org","NtpServer3":"2.nl.pool.ntp.org"}

EDIT: NTP works:
07:28:29.832 RSL: SENSOR = {"Time":"2023-08-16T07:28:29","Switch1":"ON","Switch2":"OFF","Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
**07:30:15.889 RTC: UTC 2023-08-16T05:30:16, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
07:30:16.001 RTC: Synced by NTP**
07:33:29.843 RSL: STATE = {"Time":"2023-08-16T07:33:29","Uptime":"40T12:06:59","UptimeSec":3499619,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":
I increased logging on web-console in order to see what happens but will have to wait for the children to be awake :-)

**Reply by uchristo** [reply]
I now tried with weblog 3 enabled....
The goal is to get only some slots between the elements of the curtain visible as the summer is getting some kind of reboot here in germany :-P
With the delay on stopping the curtain-movement it's quite difficult to achieve this

07:57:53.893 RSL: STATE = {"Time":"2023-08-16T07:57:53","Uptime":"41T12:08:47","UptimeSec":3586127,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"NEWWLAN","BSSId":"E0:28:6D:8A:02:E4","Channel":11,"Mode":"11n","RSSI":78,"Signal":-61,"LinkCount":84,"Downtime":"0T00:04:43"}}
07:57:53.899 RSL: SENSOR = {"Time":"2023-08-16T07:57:53","Switch1":"ON","Switch2":"OFF","Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
08:00:30.057 RSL: RESULT = {"WebLog":3}
08:00:50.798 SRC: Switch
08:00:50.802 RSL: RESULT = {"POWER1":"OFF"}
08:00:50.804 RSL: POWER1 = OFF
**>>>> Start of closing the curtain**
08:00:51.101 SRC: Switch
08:00:51.104 RSL: RESULT = {"POWER2":"ON"}
08:00:51.107 RSL: POWER2 = ON
08:00:51.109 SHT: Shtr1 Real 316000, Start 316000, Stop 0, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
08:00:51.112 RSL: RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":0,"Tilt":0}}
08:00:51.901 SHT: Shtr1 Real 299995, Start 316000, Stop 0, Dir -1, Delay 0, Rtc 0.80 [s], Freq 0, PWM 0, Tilt 0
08:00:51.905 RSL: RESULT = {"Shutter1":{"Position":95,"Direction":-1,"Target":0,"Tilt":0}}
08:00:52.856 SHT: Shtr1 Real 278655, Start 316000, Stop 0, Dir -1, Delay 0, Rtc 1.80 [s], Freq 0, PWM 0, Tilt 0
08:00:52.859 RSL: RESULT = {"Shutter1":{"Position":88,"Direction":-1,"Target":0,"Tilt":0}}
**>>>> Stop of closing the curtain**
08:00:53.055 SRC: Switch
08:00:53.059 RSL: RESULT = {"POWER2":"OFF"}
08:00:53.061 RSL: POWER2 = OFF
08:00:53.063 SHT: Stop 1 Mode 0 time 40
08:00:53.065 SHT: Shtr1 Real 274387, Start 274387, Stop 274387, Dir 0, Delay 0, Rtc 2.00 [s], Freq 0, PWM 0, Tilt 0
08:00:53.068 RSL: SHUTTER1 = 87
08:00:53.071 RSL: RESULT = {"Shutter1":{"Position":87,"Direction":0,"Target":87,"Tilt":0}}
**>>>> Start of opening the curtain**
08:00:54.126 SRC: Switch
08:00:54.130 RSL: RESULT = {"POWER1":"ON"}
08:00:54.132 RSL: POWER1 = ON
08:00:54.135 SHT: Shtr1 Real 274387, Start 274387, Stop 316000, Dir 1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
08:00:54.138 RSL: RESULT = {"Shutter1":{"Position":87,"Direction":1,"Target":100,"Tilt":0}}
08:00:54.879 SHT: Shtr1 Real 288387, Start 274387, Stop 316000, Dir 1, Delay 0, Rtc 0.75 [s], Freq 0, PWM 0, Tilt 0
08:00:54.882 RSL: RESULT = {"Shutter1":{"Position":91,"Direction":1,"Target":100,"Tilt":0}}
**>>>> Stop of opening the curtain**
08:00:55.130 SRC: Switch
08:00:55.133 RSL: RESULT = {"POWER1":"OFF"}
08:00:55.135 RSL: POWER1 = OFF
08:00:55.138 SHT: Stop 1 Mode 0 time 20
08:00:55.141 SHT: Shtr1 Real 293387, Start 293387, Stop 293387, Dir 0, Delay 0, Rtc 1.00 [s], Freq 0, PWM 0, Tilt 0
08:00:55.143 RSL: SHUTTER1 = 93
08:00:55.146 RSL: RESULT = {"Shutter1":{"Position":93,"Direction":0,"Target":93,"Tilt":0}}

**Reply by uchristo** [reply]
Can anybody help me debugging this? I'm not familiar with the values shown... :-(

**Reply by barbudor** [reply]
What the values show is that there seems to be no lag in the software
Exemple: switch action detected at 08:00:54.126 is interpreted and result in POWER and curtain within a very few milliseconds

What software logs cannot show is if there is any latency introduced by the hardawe such as a too big RC filter.

Note that Setoption132 0 has no impact on switches, only on Buttons

I suggest taking the Athom out of the wall to investigate further
If you set it up as a basic dual relay and driving lights instead of motors, do you see the same lag?

**Reply by uchristo** [reply]
Well.... this will take a while.
I doubt that the RC Filter is in a wrong dimension. They're selling this damn thing explicitly as a "curtain module"

**Comment by barbudor** [comment]
@stefanbode, as the shutter guru, you may have some ideas?

**Reply by stefanbode** [reply]
Yes for sure. Normally for switches you have to decouple them from the relay as described in the doc. Then the only thing is to decide if they are inverted or not.

Please provide a log file level 4 and give the timestamp where you physically push the switch. Currently I assume it reacts late on the hold event, but this must be checked.

The best experience you will get replacing the switch with a momentary button und use the nice shutterbutton commands. They do not work with switches.

**Reply by stefanbode** [reply]
I checked the log you provided. Agree that these look as expected. You can make the following test. Put in a webconsole the command "power1 on". Check if the shutter immediately starts. If you see here either a delay it is something in the hardware. If this works fine we have to look. Loglevel 2 should be ok actually to get important information.

**Reply by stefanbode** [reply]
Regarding the Shelly 2.5pm. I have the old 8266 based and 3 new ones based on esp32 working without any problems. I think they are sold as Shelly Plus 2PM PCB v0.1.9. Anyhow the formfactor is awesome and I really like these.

**Reply by uchristo** [reply]
I'll test the Variant with "power1 on" in the next days.

**Reply by uchristo** [reply]
Has been a while since last update, sorry for that. I got a bunch of new KM01 which I'll use as test in a "non productive environment"....

**Comment by uchristo** [comment]
The Documentation for Switchdebounce mentions Switchmodes 0....3 for handling. When I type this in (Either "switchdebounce 0" or "switchdebounce1 0"), weblog still reports "300"

11:16:44.863 CMD: switchdebounce1 0
11:16:44.868 RSL: RESULT = {"SwitchDebounce":300}
11:16:48.196 CMD: switchdebounce1 1
11:16:48.201 RSL: RESULT = {"SwitchDebounce":300}
11:16:52.841 CMD: switchdebounce 1
11:16:52.845 RSL: RESULT = {"SwitchDebounce":300}
11:18:55.441 CMD: template
11:18:55.447 RSL: RESULT = {"NAME":"Athom Curtain Module","GPIO":[0,0,0,160,161,32,0,0,224,576,225,0,0,0],"FLAG":0,"BASE":18}

Setting 40 is working indeed:
11:19:17.732 CMD: Switchdebounce 40
11:19:17.736 RSL: RESULT = {"SwitchDebounce":40}
11:19:20.889 CMD: Switchdebounce 1
11:19:20.893 RSL: RESULT = {"SwitchDebounce":40}

Reference:
<html><body>

SwitchDebounce | User control over switch debounce timing and method40..1000 = set switch debounce time in milliseconds (default = 50).  The granularity is 10 milliseconds, so the normally unnecessary last  digit is used by the debouncing code to flag special handling:  0 = no special handling  1 = force_high: only a debounce time long LOW pulse could turn the switch off  2 = force_low: only a debounce time long HIGH pulse could turn the switch on  3 = force_high + force_low  4..8 = unused  9 = AC detection for switches / relays similar to MOES MS-104B / BlitzWolf SS5 etc. If the AC frequency is 50 Hz, SwitchDebounce 69 will turn on the switch after three pulses and off after three missing one.
-- | --



</body>
</html>SwitchDebounce
	User control over switch debounce timing and method
40..1000 = set switch debounce time in milliseconds (default = 50). The granularity is 10 milliseconds, so the normally unnecessary last digit is used by the debouncing code to flag special handling:
0 = no special handling
1 = force_high: only a debounce time long LOW pulse could turn the switch off
2 = force_low: only a debounce time long HIGH pulse could turn the switch on
3 = force_high + force_low
4..8 = unused
9 = [AC detection](https://tasmota.github.io/docs/Buttons-and-Switches/#ac-frequency-detection-switch) for switches / relays similar to MOES MS-104B / BlitzWolf SS5 etc. If the AC frequency is 50 Hz, SwitchDebounce 69 will turn on the switch after three pulses and off after three missing one.

**Reply by uchristo** [reply]
shame on me
**"so the normally unnecessary last digit is used by the debouncing code to flag special handling: 0 = no special handling"**

I'll try to use 41,42,43 and test

**Reply by uchristo** [reply]
There's no noticeable difference in using 1,2,3 unfortunately... But it's also hard to tell how long the delay is. I already opened the KM01 in order to check whether there's a cap in parallel which discharges after Power off.

This is the current config:
so80 1
interlock1 1
restart 1
shutteropenduration1 15.6
shuttercloseduration1 15.0
switchmode1 2
switchmode2 2
switchdebounce1 40
switchdebounce2 40
http://ota.tasmota.com/tasmota/release/tasmota-DE.bin.gz
Backlog0 Timezone 99; TimeStd 0,0,10,1,3,60; TimeDst 0,0,3,1,2,120

**Reply by uchristo** [reply]
I took two videos.
The one with the shadows is the KM01, you can notice the "click" from the switch and the slightly delayed poweroff. The second video was taken from a shelly2.5 which responds immediately
https://drive.google.com/file/d/1rkF9CHMHz02PBkc1RU4_-wcL8MR_MWYv/view?usp=drive_link
https://drive.google.com/file/d/1BKZSBAr-QNXccQGZK5hr2UA6lWsKgzgx/view?usp=drive_link

**Reply by barbudor** [reply]
Delay/Debouncing may also comes from the hardware
I don't have this specific device so I can't tell but if the hardware include some capacitor to filter debouncing, then Tasmota can't do anything against it.
As far as the same binary is used and configured identically on the Shelly and the Athom, this seems to become the most likely conclusion.

**Reply by uchristo** [reply]
The main difference between both devices is that inputs for S1 and S2 on Athom KM01 shall be the neutral conductor (N), not phase (L).... There are no caps inside, I already disassembled the Athom

Wiring diagram (with L instead of N connected to S1 and S2) https://de.athom.tech/blank-1/tasmota-curtain-module 
![grafik](https://github.com/arendst/Tasmota/assets/27267987/64b2dba8-cf4a-49a7-bda9-423b4e656480)

**Reply by uchristo** [reply]
I tried a bit. "Switchmode 0" and "Switchmode 2" are behaving the same. I had a look at /tasmota/tasmota_support/support_switch_v4.ino in order to understand whats happening there. Seems sane.

I'll try to add debug in order to log something when the physical change of the switch is noticed.

19:57:16.961 SRC: Switch
19:57:16.963 SHT: FUNC_SET_POWER Relaymask 2 SwitchedRelay:2 by Switch, payload 1, powermask 0
19:57:16.965 SHT: Shtr1, Target 383416, Power: 0, tiltmv: 0
19:57:16.967 SHT: FUNC_SET_POWER end. powermask 0
19:57:16.971 RSL: RESULT = {"POWER2":"OFF"}
19:57:16.973 RSL: POWER2 = OFF
19:57:16.976 SHT: Shtr1 Time 19(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 383416, nStop 0, Trgt 383416, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:16.978 SHT: Stop 1 Mode 0 time 19
19:57:16.981 SHT: Shtr1 Real 383416, Start 383416, Stop 383416, Dir 0, Delay 0, Rtc 0.95 [s], Freq 0, PWM 0, Tilt 0
19:57:16.984 RSL: SHUTTER1 = 88
19:57:16.986 RSL: RESULT = {"Shutter1":{"Position":88,"Direction":0,"Target":88,"Tilt":0}}
19:57:18.716 SRC: Switch
19:57:18.718 SHT: FUNC_SET_POWER Relaymask 0 SwitchedRelay:2 by Switch, payload 1, powermask 2
19:57:18.720 SHT: Delay Start? var1 <99>=<>, max10s?
19:57:18.722 SHT: Shtr1, Target 0, Power: 2, tiltmv: 0
19:57:18.723 SHT: FUNC_SET_POWER end. powermask 2
19:57:18.726 RSL: RESULT = {"POWER2":"ON"}
19:57:18.729 RSL: POWER2 = ON
19:57:18.732 SHT: Shtr1 Real 383416, Start 383416, Stop 0, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
19:57:18.734 RSL: RESULT = {"Shutter1":{"Position":88,"Direction":-1,"Target":0,"Tilt":0}}
19:57:18.737 SHT: Shtr1 Time 0(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 383416, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:18.767 SHT: Shtr1 Time 1(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 383416, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:18.893 SHT: Shtr1 Time 3(1), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 381264, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:18.949 SHT: Shtr1 Time 4(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 380188, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:18.999 SHT: Shtr1 Time 5(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 379112, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.048 SHT: Shtr1 Time 6(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 378036, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.098 SHT: Shtr1 Time 7(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 376960, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.148 SHT: Shtr1 Time 8(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 375884, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.200 SHT: Shtr1 Time 9(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 374808, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.214 SHT: Shtr1 Real 373732, Start 383416, Stop 0, Dir -1, Delay 0, Rtc 0.50 [s], Freq 0, PWM 0, Tilt 0
19:57:19.217 RSL: RESULT = {"Shutter1":{"Position":84,"Direction":-1,"Target":0,"Tilt":0}}
19:57:19.250 SHT: Shtr1 Time 10(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 373732, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.300 SHT: Shtr1 Time 11(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 372656, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.350 SHT: Shtr1 Time 12(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 371580, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.400 SHT: Shtr1 Time 13(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 370504, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.449 SHT: Shtr1 Time 14(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 369428, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.500 SHT: Shtr1 Time 15(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 368352, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.550 SHT: Shtr1 Time 16(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 367276, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.600 SHT: Shtr1 Time 17(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 366200, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.649 SHT: Shtr1 Time 18(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 365124, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.700 SRC: Switch
19:57:19.702 SHT: FUNC_SET_POWER Relaymask 2 SwitchedRelay:2 by Switch, payload 1, powermask 0
19:57:19.704 SHT: Shtr1, Target 364048, Power: 0, tiltmv: 0
19:57:19.706 SHT: FUNC_SET_POWER end. powermask 0
19:57:19.709 RSL: RESULT = {"POWER2":"OFF"}
19:57:19.711 RSL: POWER2 = OFF
19:57:19.714 SHT: Shtr1 Time 19(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 364048, nStop 0, Trgt 364048, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
19:57:19.717 SHT: Stop 1 Mode 0 time 19
19:57:19.720 SHT: Shtr1 Real 364048, Start 364048, Stop 364048, Dir 0, Delay 0, Rtc 0.95 [s], Freq 0, PWM 0, Tilt 0
19:57:19.722 RSL: SHUTTER1 = 81
19:57:19.725 RSL: RESULT = {"Shutter1":{"Position":81,"Direction":0,"Target":81,"Tilt":0}}
19:57:24.015 WIF: Prüfe Verbindung…
19:57:33.190 WIF: Sending Gratuitous ARP
19:57:34.200 RSL: STATE = {"Time":"2024-01-09T19:57:34","Uptime":"1T00:35:04","UptimeSec":88504,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"NEWWLAN","BSSId":"98:9B:CB:FB:CE:F5","Channel":6,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":2,"Downtime":"0T00:00:11"}}
19:57:34.205 RSL: SENSOR = {"Time":"2024-01-09T19:57:34","Switch1":"OFF","Switch2":"ON","Shutter1":{"Position":81,"Direction":0,"Target":81,"Tilt":0}}
19:57:44.015 WIF: Prüfe Verbindung…

**Reply by stefanbode** [reply]
Thanks for sharing the logs.between 19:57:18.716 SRC: Switch on the command to power on the relay there is only 0.15sec. This is typical.

**Reply by stefanbode** [reply]
Same with the power off. Also here the system power off just 0.1 sec after the switch is detected. Therefore I would guess we may have an issue with the switch and the time it takes to report ist to the rest of the system.  Do not see a direct issue in TASMOTA. I must admit I replaced my shutter switches with buttons. Much more convenient and looks more professional. Additionally you can use this nice shutterbutton options. But for sure at an additional cost. From far away my candidate would be the debounce either. I would play with different values and see if the lag changes.

**Reply by uchristo** [reply]
I took some pics from the Hardware:
https://drive.google.com/file/d/1kUmpP7tBeWDVnDNtg4BH9JqhBBk15x-7/view?usp=drive_link
https://drive.google.com/file/d/1wpoJLjvDql1B_Q1R_hg4lxA-2XXkfxAa/view?usp=sharing
https://drive.google.com/file/d/1HpDsh0-vRUQY2IgfYe0iVR1ZJUTU1eeX/view?usp=sharing

Default Debounce for this HW is 300 milliseconds. When reduced to 40, the PowerON does work immediately, but PowerOFF has lag (see in Videos, they are visible for everybody now)

**Reply by stefanbode** [reply]
Thanks for granting access to the video. It's always better than thousands of words. Indeed the power on works directly. And there is a small delay on off. The way you used the shutter driver you still control the relay's directly with the switches. The driver kicks in after the relay has stopped already. This is at least my current believe. Therefore the delay must be in the switches. If you change the switchmode to it inverts high and low. Do you still have the delay on stop or does it now act on open as I would expect. ?

**Reply by stefanbode** [reply]
I just forget that I took my testbench with me on holiday. So easy debug now. Your issue is the switchdebounce. With the default 50 everything is fine. I was able to reproduce with switchbounce 300.

**Reply by uchristo** [reply]
Well, even with Default "50" the lag is still noticeable, but I'll learn to live with it :-)
Migrated all 12 shutters from Shelly 2.5 to Athom KM01 now. I'm currently using switchmode 2 because the athoms are triggered by N. Tried 0 and 1 aswell (1 was configured on the old shellies).
